lib/mac/CMakeLists.txt

```diff
@@ -53,44 +57,55 @@ if (ACACIA_PYTHON OR ACACIA_NODEJS)
   include(UseSWIG)
 endif()
 
-# SWIG Instructions to build a python module called "acacia_axapi"
+# Build and install a python module called "acacia_axapi" using setuptools
 
 if (ACACIA_PYTHON)
-  find_package(Python3 COMPONENTS Interpreter Development)
-
-  include_directories(${CMAKE_CURRENT_SOURCE_DIR})
-  include_directories(${Python3_INCLUDE_DIRS})
+  configure_file(python/setup.py.in python/setup.py)
+  configure_file(python/pyproject.toml python/pyproject.toml)
+  configure_file(acacia_axapi.i python/acacia_axapi/acacia_axapi.i)
 
-  set_property(SOURCE acacia_axapi.i PROPERTY CPLUSPLUS ON)
+  find_package(Python COMPONENTS Interpreter Development)
 
-  swig_add_library(
-    # The name of the c++ library used by python module
-    python_acacia_axapi
-    TYPE SHARED
-    LANGUAGE python
-    SOURCES acacia_axapi.i
+  add_custom_target(
+    python_acacia_axapi ALL
+    DEPENDS
+      acacia_axapi
+      python/setup.py
+    WORKING_DIRECTORY
+      python
+    COMMAND
+      MACOSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET} ${Python_EXECUTABLE} -m build .
   )
 
-  set_target_properties(
-    python_acacia_axapi
-
-    PROPERTIES
-    SUFFIX .so
+  set_property(
+    TARGET python_acacia_axapi
+    APPEND
+    PROPERTY ADDITIONAL_CLEAN_FILES
+      ${CMAKE_CURRENT_BINARY_DIR}/python/__pycache__
+      ${CMAKE_CURRENT_BINARY_DIR}/python/acacia_axapi.egg-info
+      ${CMAKE_CURRENT_BINARY_DIR}/python/dist
+      ${CMAKE_CURRENT_BINARY_DIR}/python/acacia_axapi.py
+      ${CMAKE_CURRENT_BINARY_DIR}/python/acacia_axapi_wrap.cpp
   )
 
-  target_link_libraries(
-    # Target to link
-    python_acacia_axapi
-
-    # Items to link into the target
-    acacia_axapi
-    ${Python3_LIBRARIES}
+  add_custom_target(
+    rename_init_py
+    DEPENDS
+      python_acacia_axapi
+    WORKING_DIRECTORY
+      python/acacia_axapi
+    COMMAND
+      ${CMAKE_COMMAND} -E rename acacia_axapi.py __init__.py
```

Comment:
whoa, what is going on here??

Reply:
Ugh, yeah, I should have left a comment about this. This is the fix for the `from acacia_axapi import acacia_axapi` thing - `__init__.py` is like the "top level" of the module, so it moves everything in `acacia_api.py` up to the top instead of one level down. 

This does feel really hacky, but I think it's reasonable since we really want the SWIG-generated package to be the top level for now (perhaps someday we'll write Python wrapper code and make _that_ the top level).

Added a comment.